### PR TITLE
Extract foo library into a separate library

### LINKED ISSUE #907: Extract foo_library into a separate library
[foo_library](https://github.com/open-telemetry/opentelemetry-cpp/tree/main/examples/otlp/foo_library) was copied into multiple examples which requires extra effort to keep the copies in sync. It could be extracted to a separated library for all examples to link.

### PR DESCRIPTION
**Pull request by esigo**
Fixes #907 (issue)

## Changes

Please provide a brief description of the changes here.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Unit tests have been added
* [ ] Changes in public API reviewed

### CODE REVIEW

File: examples/CMakeLists.txt [hunk 1]
```diff
@@ -1,3 +1,4 @@
+add_subdirectory(common)
```
Comment (lalitb) [inline examples/CMakeLists.txt:1]: Nit: we can also add ` include_directories(common)`, and remove the dependencies from subfolders ?

File: examples/common/foo_library/BUILD [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    srcs = [
+        "foo_library.cc",
+    ],
+    hdrs = [
+        "foo_library.h",
+    ],
+    defines = ["BAZEL_BUILD"],
+    deps = [
+        "//api",
+        "//sdk/src/trace",
```
Comment (lalitb) [inline examples/common/foo_library/BUILD:14]: Does it have sdk dependency?

**Review by lalitb — APPROVED** [review]
LGTM with minor comments. Thanks for the PR.

File: examples/CMakeLists.txt (hunk 1 above)
Comment (esigo) [reply examples/CMakeLists.txt:1]: good one, thanks

File: examples/common/foo_library/BUILD (hunk 2 above)
Comment (esigo) [reply examples/common/foo_library/BUILD]: it needes sdk:headers, just fixed it. thanks

**Review by ThomsonTan — APPROVED** [review]

**Pull request merged**